### ISSUE TITLE
[unbound-method]: don't report unbound methods passed to `jest.mocked`

### ISSUE THREAD

**Issue by FloEdelmann** [jest-community/eslint-plugin-jest#1596]
```ts
import foo from './foo'

jest.mock('./foo')

jest.mocked(foo.bar).mockResolvedValue(baz)
//          ^^^^^^^ jest/unbound-method reports an error here

expect(foo.bar).toHaveBeenCalled() // no error here
```

Since `jest.mocked` will be called with mock functions, it should not make a difference whether it is bound or not, right? So I'd suggest to not report an error for unbound methods passed to `jest.mocked`, just like already happens when passing it to `expect`.
